pkg/plugin/processor/builtin/unwrap_opencdc.go

```diff
@@ -0,0 +1,290 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package builtin
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	"github.com/conduitio/conduit-commons/config"
+	"github.com/conduitio/conduit-commons/opencdc"
+	sdk "github.com/conduitio/conduit-processor-sdk"
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+	"github.com/conduitio/conduit/pkg/foundation/log"
+	"github.com/goccy/go-json"
+)
+
+type unwrapOpenCDC struct {
+	sdk.UnimplementedProcessor
+
+	logger      log.CtxLogger
+	fieldRefRes sdk.ReferenceResolver
+}
+
+func newUnwrapOpenCDC(logger log.CtxLogger) sdk.Processor {
+	return &unwrapOpenCDC{logger: logger}
+}
+
+func (u *unwrapOpenCDC) Specification() (sdk.Specification, error) {
+	return sdk.Specification{
+		Name:    "unwrap.opencdc",
+		Summary: "A processor that unwraps the OpenCDC record saved in one of record's fields.",
+		Description: `
+The unwrap.cdc processors unwraps the OpenCDC record saved in one of record's fields.
+This is useful in situations where a record goes through intermediate systems before 
+being written to a final destination. In these cases, the original OpenCDC record is
+part of the payload read from the intermediate system and needs to be unwrapped before
+being written to the destination.
+'`,
+		Version: "v0.1.0",
+		Author:  "Meroxa, Inc.",
+		Parameters: config.Parameters{
+			"field": config.Parameter{
+				Default: ".Payload.After",
+				Description: `
+A reference to the field which contains the OpenCDC record.
+
+For more information about record references, see: https://github.com/ConduitIO/conduit-processor-sdk/blob/cbdc5dcb5d3109f8f13b88624c9e360076b0bcdb/util.go#L66.`,
+				Type: config.ParameterTypeString,
+			},
+		},
+	}, nil
+}
+
+func (u *unwrapOpenCDC) Configure(_ context.Context, m map[string]string) error {
+	field, ok := m["field"]
+	if !ok {
+		return cerrors.New("missing required parameter 'field'")
+	}
+
+	if !strings.HasPrefix(field, ".Payload") {
+		return cerrors.Errorf("only payload can be unwrapped, field given: %v", field)
+	}
+
+	rr, err := sdk.NewReferenceResolver(field)
+	if err != nil {
+		return cerrors.Errorf("invalid reference: %w", err)
+	}
+
+	u.fieldRefRes = rr
+	return nil
+}
+
+func (u *unwrapOpenCDC) Open(context.Context) error {
+	return nil
+}
+
+func (u *unwrapOpenCDC) Process(_ context.Context, records []opencdc.Record) []sdk.ProcessedRecord {
+	out := make([]sdk.ProcessedRecord, len(records))
+	for i, rec := range records {
+		out[i] = u.processRecord(rec)
```

Comment:
I think we talked about this slice in my processor's PR, should be better to initialize it and append records.
we should also make sure to return the slice if an error is returned for any record by `u.processRecord`

Reply:
Agreed!